presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -307,6 +311,19 @@ private void processPage(Page page)
             for (Aggregator aggregator : aggregators) {
                 aggregator.processPage(groupIds, page);
             }
+
+            recordHashCollisions();
```

Comment:
You might be right. I have moved `recordHashCollisions` to `getOutput` function. Now hash collisions are recorded when `aggregationBuilder` is nullified (e.g. when it is full).